### ISSUE TITLE
Should Completable.zip have a resultsSelector?

### ISSUE THREAD

**Issue by scottrhoyt** [ReactiveX/RxSwift#1245]
**Short description of the issue**:

  I would like to create a `Completable` by combining two source `Completable` so that the combined `Completable` completes only when both sources have completed.

**Expected outcome**:

  I would expect that I should be able to accomplish this like this:

```swift
let combinedCompletable = Completable.zip(completable1, completable2)
```

**What actually happens**:

I need to provide a throw away `resultsSelector` like this:

```swift
let combinedCompletable = Completable.zip(completable1, completable2) { _ in throw SomeError }
```

**Self contained code example that reproduces the issue**:

_see above_

**RxSwift/RxCocoa/RxBlocking/RxTest version/commit**

  _3.4.1_

**Platform/Environment**

  - [x] iOS
  - [ ] macOS
  - [ ] tvOS
  - [ ] watchOS
  - [ ] playgrounds

**How easy is to reproduce? (chances of successful reproduce after running the self contained code)**

  - [x] easy, 100% repro
  - [ ] sometimes, 10%-100%
  - [ ] hard, 2% - 10%
  - [ ] extremely hard, %0 - 2%

**Xcode version**:

```
  Xcode version goes here
```

:warning: Fields below are optional for general issues or in case those questions aren't related to your issue, but filling them out will increase the chances of getting your issue resolved. :warning:

**Installation method**:
  - [ ] CocoaPods
  - [x] Carthage
  - [ ] Git submodules

**I have multiple versions of Xcode installed**:
  (so we can know if this is a potential cause of your issue)
  - [x] yes (8.3.1, 8.2.1, 7.3)
  - [ ] no

**Level of RxSwift knowledge**:
  (this is so we can understand your level of knowledge
    and formulate the response in an appropriate manner)
  - [ ] just starting
  - [ ] I have a small code base
  - [x] I have a significant code base

**Comment by kzaher** [comment]
Hi @scottrhoyt ,

yeah, we need to improve this.

**Comment by kzaher** [comment]
Hi @scottrhoyt ,

we've merged #1251 yesterday. You can use `merge` for now. We will probably also add a `zip` alias in RxSwift 4.0.

**Comment by scottrhoyt** [comment]
Thanks @kzaher !
